### PR TITLE
api: make displaytext form upload template/iso optional

### PR DESCRIPTION
**Pull request by shwstppr**
### Description

displaytext parameter for registerTemplate/Iso API has been optional sometime back. For upload APIs the param is not optional which creates inconsistency in the UI. This PR fixes the behaviour.

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)
- [ ] build/CI

#### Feature/Enhancement Scale

- [ ] Major
- [ ] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [ ] Minor
- [ ] Trivial

### Screenshots (if appropriate):
Before change,
Screenshot from 2023-12-01 12-54-10

### CODE REVIEW

File: api/src/main/java/org/apache/cloudstack/api/command/user/iso/GetUploadParamsForIsoCmd.java [hunk 1]
```diff
@@ -85,7 +85,7 @@ public Boolean isBootable() {
     }
 
     public String getDisplayText() {
-        return displayText;
+        return StringUtils.isEmpty(displayText) ? getName() : displayText;
```
Comment (weizhouapache) [inline api/src/main/java/org/apache/cloudstack/api/command/user/iso/GetUploadParamsForIsoCmd.java:88]: may `isBlank` be better than `isEmpty` ?

Comment (shwstppr) [reply api/src/main/java/org/apache/cloudstack/api/command/user/iso/GetUploadParamsForIsoCmd.java]: done

**Review by weizhouapache — APPROVED** [review]
code lgtm

**Comment by shwstppr** [comment]
@blueorangutan package

**Review by DaanHoogland — APPROVED** [review]
clgtm, great! let's remove the requirement for display names and - texts where ever a name is available as well.

**Pull request merged**